infrastructure/lib/api.ts

```diff
@@ -0,0 +1,136 @@
+        name: PARTITION_KEY_NAME,
+        type: AttributeType.STRING
+      },
+      tableName: 'monitors',
+
+      // The default removal policy is RETAIN, which means that cdk destroy will not attempt to delete
+      // the new table, and it will remain in your account until manually deleted. By setting the policy to
+      // DESTROY, cdk destroy will delete the table (even if it has data in it)
+      removalPolicy: RemovalPolicy.DESTROY, // NOT recommended for production code
+    });
+
+    // Define the Lambdas to access the DynamoDB table
+
+    const getAll = new Function(this, 'getAllMonitorsFunction', {
+      code: new AssetCode(join(__dirname, 'functions')),
+      handler: 'get-all.handler',
+      runtime: Runtime.NODEJS_10_X,
+      environment: {
+        TABLE_NAME: this.table.tableName,
+        PRIMARY_KEY: PARTITION_KEY_NAME
+      },
+      initialPolicy: [
+        new PolicyStatement({
+          actions: ['cloudwatch:DescribeAlarms'],
+          resources: ['*'],
+        })
+      ]
+    });
+
+    const createOne = new Function(this, 'createMonitorFunction', {
+      code: new AssetCode(join(__dirname, 'functions')),
+      handler: 'create.handler',
+      runtime: Runtime.NODEJS_10_X,
+      environment: {
+        TABLE_NAME: this.table.tableName,
+        PRIMARY_KEY: PARTITION_KEY_NAME
+      },
+      initialPolicy: [
+        new PolicyStatement({
+          actions: ['cloudwatch:PutMetricAlarm'],
+          resources: ['*'],
+        })
+      ]
+    });
+
+    const updateOne = new Function(this, 'updateMonitorFunction', {
+      code: new AssetCode(join(__dirname, 'functions')),
+      handler: 'update.handler',
+      runtime: Runtime.NODEJS_10_X,
+      environment: {
+        TABLE_NAME: this.table.tableName,
+        PRIMARY_KEY: PARTITION_KEY_NAME
+      }
+    });
+
+    const deleteOne = new Function(this, 'deleteMonitorFunction', {
+      code: new AssetCode(join(__dirname, 'functions')),
+      handler: 'delete.handler',
+      runtime: Runtime.NODEJS_10_X,
+      environment: {
+        TABLE_NAME: this.table.tableName,
+        PRIMARY_KEY: PARTITION_KEY_NAME
+      },
+      initialPolicy: [
+        new PolicyStatement({
+          actions: ['cloudwatch:DeleteAlarms'],
+          resources: ['*']
+        })
+      ]
+    });
+
+    this.table.grantReadWriteData(getAll);
+    this.table.grantReadWriteData(createOne);
+    this.table.grantReadWriteData(updateOne);
+    this.table.grantReadWriteData(deleteOne);
+
+    // Create the API to access the monitors
+
+    this.api = new RestApi(this, 'SHS API', {
+      restApiName: 'Service Health System',
+      description: 'This service registers and fetches monitors and their statuses.',
+    });
+
+    const status = this.api.root.addResource('status');
+    const getStatusIntegration = new LambdaIntegration(getAll);
+    status.addMethod('GET', getStatusIntegration);
+    status.addCorsPreflight({allowOrigins: ['*'], allowMethods: ['GET']});
```

Comment:
you should be able to specify `defaultCors` at the API level so all of it will be permitted